charmcraft/helptexts.py

```diff
@@ -265,7 +265,7 @@ def get_command_help(self, command, arguments):
 
         # recommend other commands of the same group
         for group_name, _, command_classes in self.command_groups:
-            if group_name == command.group:
+            if any(isinstance(command, command_class) for command_class in command_classes):
```

Comment:
This was the real only place where the `group` attribute was used: to find the command group for a given command, when presenting help. 

The new code iterates on all the commands, but this is not a problem as the list will always be small.